src/Controller/GameController.php

```diff
@@ -17,12 +21,70 @@ public function __construct(GameManipulator $gameManipulator)
         $this->gameManipulator = $gameManipulator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function random(Request $request, Game $game): Response
+    {
+        $triggers = [];
+
+        foreach (range(0, 9) as $x) {
+            foreach (range(0, 9) as $y) {
+                $triggers[] = $this->createForm(ShootShipType::class, [
+                    'game_hash' => $game->getHash(),
+                    'coordinates' => join(",", [$x, $y])
+                ]);
+            }
+        }
+
+        $trigger = $this->createForm(ShootShipType::class)->handleRequest($request);
+
+        if ($trigger->isSubmitted() && $trigger->isValid()) {
+            $data = $trigger->getData();
+
+            $game = $this->getDoctrine()
```

Comment:
You can use an EntityType inside the ShootShipType for "game" attribute (instead of "game_hash").

This way you'll directly have your game object without using the repository.

Moreover, form validation will fail automatically if the game doesn't exist

Reply:
Will do later